index.js

```diff
@@ -2,6 +2,7 @@
 
 module.exports = {
   rules: {
-    'accessing-ember-namespace': require('./lib/rules/accessing-ember-namespace')
+    'accessing-ember-namespace': require('./lib/rules/accessing-ember-namespace'),
+    'disallow-function-prototype-extension': require('./lib/rules/disallow-function-prototype-extension')
```

Comment:
Yeah, maybe `no-function-prototype-extension-calls` and `desctructure-ember-namespace`. I really don't like using the word `destructure` in there because this is perfectly valid:

```javascript
run = Ember.run;
run.bind(...);
```

But I can't think of a concise way to describe that precisely and I think in practice people will think of this as requiring destructuring as an imitation of module imports.

Interesting to note, @bantic, that this would not be allowed: 

```javascript
import Ember from 'ember';

export default Ember.Component.extend();
```

We could add exceptions for commonly used factories, but technically it does not fit the rule.